yt_dlp/extractor/livestream.py

```diff
@@ -207,21 +223,39 @@ def _extract_event(self, event_data):
                     'http://livestream.com/accounts/%s/events/%s/videos/%s' % (account_id, event_id, v_id),
                     'Livestream', v_id, v.get('caption')))
             last_video = videos_info[-1]['id']
+        return entries
+
+    def _extract_event(self, event_data):
+        event_id = compat_str(event_data['id'])
+        entries = self._generate_event_playlist(event_data)
         return self.playlist_result(entries, event_id, event_data['full_name'])
 
+    def _extract_account_events(self, account_data):
+        upcoming_events = account_data['upcoming_events']
+        past_events = account_data['past_events']
+        entries = []
+        events = upcoming_events['data'] + past_events['data']
+        for event_data in events:
+            playlist = self._generate_event_playlist(event_data)
+            if playlist:
+                entries = entries + playlist
+        return self.playlist_result(entries, account_data['id'], account_data['full_name'])
```

Comment:
```suggestion
    def _extract_account_events(self, account_data):
        items = traverse_obj(account_data, (('upcoming_events', 'past_events'), 'data', ...))
        return self.playlist_result(
            itertools.chain.from_iterable(map(self._generate_event_playlist, items),
            account_data.get('id'), account_data.get('full_name'))
```
Don't make this a function either

Reply:
@pukkandan I can get rid of the functions and move the logic to _real_extract, to make it look like

```python3
def _real_extract(self, url):
    mobj = self._match_valid_url(url)
    video_id = mobj.group('id')
    event = mobj.group('event_id') or mobj.group('event_name')
    account = mobj.group('account_id') or mobj.group('account_name')
    api_url = f'http://livestream.com/api/accounts/{account}'
    if video_id:
        video_data = self._download_json(
            f'{api_url}/events/{event}/videos/{video_id}', video_id)
        return self._extract_video_info(video_data)
    elif event:
        event_data = self._download_json(f'{api_url}/events/{event}', None)
        event_id = str(event_data['id'])
        entries = self._generate_event_playlist(event_data)
        return self.playlist_result(entries, event_id, event_data['full_name'])
    account_data = self._download_json(api_url, None)
    items = traverse_obj(account_data, (('upcoming_events', 'past_events'), 'data', ...))
    return self.playlist_result(
        itertools.chain.from_iterable(map(self._generate_event_playlist, items)),
        account_data.get('id'), account_data.get('full_name'))
```